openedx/core/djangoapps/content_libraries/api.py

```diff
@@ -373,20 +342,43 @@ def get_library(library_key):
     Raises ContentLibraryNotFound if the library doesn't exist.
     """
     ref = ContentLibrary.objects.get_by_key(library_key)
-    bundle_metadata = get_bundle(ref.bundle_uuid)
-    lib_bundle = LibraryBundle(library_key, ref.bundle_uuid, draft_name=DRAFT_NAME)
-    num_blocks = len(lib_bundle.get_top_level_usages())
-    last_published = lib_bundle.get_last_published_time()
-    (has_unpublished_changes, has_unpublished_deletes) = lib_bundle.has_changes()
+    learning_package = ref.learning_package
+    num_blocks = publishing_api.get_all_drafts(learning_package.id).count()
+    last_publish_log = publishing_api.get_last_publish(learning_package.id)
+    has_unpublished_changes = publishing_api.get_entities_with_unpublished_changes(
+                                  learning_package.id
+                              ).exists()
+
+    # I'm doing this one to match what was already existing, but I'm not clear
+    # on the use case for it.
+    has_unpublished_deletes = publishing_api.get_entities_with_unpublished_deletes(
+                                  learning_package.id
+                              ).exists()
+
+    # Learning Core doesn't really have a notion of a global version number,but
+    # we can sort of approximate it by using the primary key of the last publish
+    # log entry, in the sense that it will be a monotonically increasing
+    # integer, though there will be large gaps. We use 0 to denote that nothing
+    # has been done, since that will never be a valid value for a PublishLog pk.
+    #
+    # That being said, we should figure out if we really even want to keep a top
+    # level version indicator for the Library as a whole. In the v1 libs
+    # implemention, this served as a way to know whether or not there was an
+    # updated version of content that a course could pull in. But more recently,
+    # we've decided to do those version references at the level of the
+    # individual blocks being used, since a Learning Core backed library is
+    # intended to be used for many LibraryContentBlocks and not 1:1 like v1
+    # libraries
```

Comment:
Right now, we need it because the `LibraryContentBlock` wants something to put for the library's version, and I figure that we might as well pass it through the REST API as well. But yeah, another one for the longer term chopping block.

Reply:
I added this to our TODO list. If the library-wide version is on the chopping block, I want to get this out of the REST API before anyone starts relying on it :)